`Communication`: Enhance messages and threads user interface

egekurt123 opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
The primary motivation behind these updates is to enhance the overall user experience within the messages/communication module. By implementing these changes, we aim to achieve better usability, modernized look and feel



### Steps for Testing
<!-- Please describe in detail how reviewers can test your changes. Make sure to take all related features and views into account! Below is an example that you can refine. -->

Prerequisites:
- 1 Student / Instructor / Admin
- 1 Course with Messages and Communications enabled

1. Log in to Artemis
2. Navigate to Messages
3. Make sure everything work as expected
4. Navigate to Communications
5. Make sure everything work as expected

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)

### Review Progress
<!-- Each PR should be reviewed by at least two other developers. The code, the functionality (= manual test) and the exam mode need to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->
<!-- All PRs that might affect the exam mode (e.g. change a client component that is also used in the exam mode) need an additional verification that the exam mode still works. -->

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- The line coverage must be above 90% for changes files and you must use extensive and useful assertions for server tests and expect statements for client tests. -->
<!-- Note: Use the table below and confirm in the last column that you have implemented extensive assertions for server tests and expect statements for client tests. -->
<!--       You can use `supporting_script/generate_code_cov_table/generate_code_cov_table.py` to automatically generate one from the corresponding Bamboo build plan artefacts. -->
<!--       Remove rows with only trivial changes from the table. -->
<!--
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ExerciseService.java | 85% | ✅                           |
| programming-exercise.component.ts | 95% | ✅              |
-->

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| course-management-tab-bar.component.ts | 93.97% | ✅ ❌ |
| course.model.ts | 100% | ✅ |
| exam-update.component.ts | 88.55% | ✅ ❌ |
| exercise-title-channel-name.component.ts | 100% | ✅ |
| lecture-title-channel-name.component.ts | 100% | ✅ |
| course-conversations.component.ts | 95.12% | ✅ ❌ |
| conversation-header.component.ts | 96.36% | ✅ ❌ |
| conversation-messages.component.ts | 94.11% | ✅ ❌ |
| metis-conversation.service.ts | 84.97% | ✅ ❌ |
| metis.service.ts | 88.47% | ✅ ❌ |
| post.component.ts | 94.91% | ✅ ❌ |
| post-footer.component.ts | 100% | ✅ |
| answer-post-header.component.ts | 96.15% | ✅ ❌ |
| post-header.component.ts | 95.65% | ✅ ❌ |
| posting-header.directive.ts | 100% | ✅ |
| posting-markdown-editor.component.ts | 97.56% | ✅ ❌ |
| post-reactions-bar.component.ts | 98.33% | ✅ ❌ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| CourseRepository.java | 76% | ✅ ❌ |
| ChannelResource.java | 91% | ✅ ❌ |
| ConversationManagementResource.java | 100% | ✅ |
| ConversationResource.java | 92% | ✅ ❌ |

�

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. Remove the section if you did not change the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
#### Light Mode

Old 
<img width="1337" alt="Screenshot 2024-05-31 at 14 30 41" src="https://github.com/ls1intum/Artemis/assets/69898283/c8e97b08-632b-4f38-bca9-73a20244a143">
<img width="426" alt="Screenshot 2024-05-31 at 14 29 31" src="https://github.com/ls1intum/Artemis/assets/69898283/c4ef8615-486a-4647-9e18-ab4e39214926">

New
![Screenshot 2024-07-08 at 16 09 55](https://github.com/ls1intum/Artemis/assets/69898283/9a24c7d8-a7b6-4250-b7cf-8733727524c5)
![Screenshot 2024-07-08 at 16 14 39](https://github.com/ls1intum/Artemis/assets/69898283/a2dfffc0-4109-4a10-bd92-9fe6579b4f4c)



#### Dark Mode

Old
![Screenshot 2024-07-08 at 15 09 39](https://github.com/ls1intum/Artemis/assets/69898283/c887536b-1684-428b-a511-46a0d74871b0)
![Screenshot 2024-07-08 at 15 08 34](https://github.com/ls1intum/Artemis/assets/69898283/02b21760-575d-4602-91d9-668a4a8f6a04)

New
![Screenshot 2024-07-08 at 16 10 01](https://github.com/ls1intum/Artemis/assets/69898283/cbf8b86b-c4e8-4e48-9005-503beab0de21)
![Screenshot 2024-07-08 at 16 14 46](https://github.com/ls1intum/Artemis/assets/69898283/0bcff884-ce92-4344-9e0c-6b6698afb9e0)




### Notes:
- Option buttons like edit or delete message, as well as reactions will be updated in a seperate follow-up PR 




<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced post management with sorting and tagging functionalities.
  - Improved layout and conditional rendering in post and post footer components.
  - Added dynamic display for answer posts based on conditions.

- **Bug Fixes**
  - Corrected the display issues in various post components.
  
- **Style**
  - Updated badge styles with role-specific colors for students, tutors, and instructors.

- **Refactor**
  - Renamed function for clarity in communication settings within the exam update component.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

az108 reviewed (dismissed):
Tested on TS4 as Instructor and Student. Works as expected and I like the changes a lot 👍 . Code LGTM aswell!

Jan-Thurner reviewed (approved):
Tested on TS1 as an instructor and as a Student. Though the functionality works as expected, and the UI looks loads better than before, I have a few remarks:

The Distance between the two buttons on the top right feels a bit big:
![image](https://github.com/ls1intum/Artemis/assets/107639007/278b4fc2-4d49-44d3-8a5a-ab51b2dbdd55)
If this view is opened on a small screen in combination with opening in thread answers, the right button disappears behind the threads:
![image](https://github.com/ls1intum/Artemis/assets/107639007/f52f7dd6-4c52-481d-b74a-f6d253290212)

The location of the `Send` button at the bottom right feels a bit odd to me too. The gap between the button and the textbox is rather big and the button is touching the right border of the window.
![image](https://github.com/ls1intum/Artemis/assets/107639007/63b35ee5-5a2e-4ca1-8b83-3787024f52a2)

As the UI improvements are great and working, I think the suggestions can be moved into a followup PR.

sarpsahinalp reviewed (approved):
code lgtm

edkaya reviewed (approved):
code lgtm

egekurt123 commented:
> Tested on TS1 as an instructor and as a Student. Though the functionality works as expected, and the UI looks loads better than before, I have a few remarks:
> 
> The Distance between the two buttons on the top right feels a bit big: ![image](https://private-user-images.githubusercontent.com/107639007/335826505-278b4fc2-4d49-44d3-8a5a-ab51b2dbdd55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MDM4ODEsIm5iZiI6MTcxNzQwMzU4MSwicGF0aCI6Ii8xMDc2MzkwMDcvMzM1ODI2NTA1LTI3OGI0ZmMyLTRkNDktNDRkMy04YTVhLWFiNTFiMmRiZGQ1NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QwODMzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNWQwYzY3ZDJhMWY4NjhlMmY0NWUxNmRlMjNiOTEwMzI3YzA3MTIyZTRkMTUzNGI0N2VhNjJlNzQyYWZhNDFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aIfsxIeAJSozv6_AR4ye9rJVh21nVzYeReppEXauk1k) If this view is opened on a small screen in combination with opening in thread answers, the right button disappears behind the threads: ![image](https://private-user-images.githubusercontent.com/107639007/335826556-f52f7dd6-4c52-481d-b74a-f6d253290212.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MDM4ODEsIm5iZiI6MTcxNzQwMzU4MSwicGF0aCI6Ii8xMDc2MzkwMDcvMzM1ODI2NTU2LWY1MmY3ZGQ2LTRjNTItNDgxZC1iNzRhLWY2ZDI1MzI5MDIxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QwODMzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZmM3Y2Y4YmU2Mjc4ZmExNGJhNDU3YzQ4M2I3NDk0ZWU0ODQwYWIzNTVmZmQ2MDNlZWZmODVmYWNkNzY2ZjZmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QExPNumCLa5fpSnyOZbvulSIhjXel8L2zlWkTyACkHs)
> 
> The location of the `Send` button at the bottom right feels a bit odd to me too. The gap between the button and the textbox is rather big and the button is touching the right border of the window. ![image](https://private-user-images.githubusercontent.com/107639007/335826686-63b35ee5-5a2e-4ca1-8b83-3787024f52a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0MDM4ODEsIm5iZiI6MTcxNzQwMzU4MSwicGF0aCI6Ii8xMDc2MzkwMDcvMzM1ODI2Njg2LTYzYjM1ZWU1LTVhMmUtNGNhMS04YjgzLTM3ODcwMjRmNTJhMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QwODMzMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTMxYTM4YTlkN2ZkNWU1ZmM5MTM5NzM2ZTA0ZmVlYWM0YmQ4NzhiZmM2N2UzYjcwYWFjYjc3NTU4M2E2NWJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ke0Qb7DwIVz28rS4MNoltvwxF9pJ14HAEZtntd5h4x0)
> 
> As the UI improvements are great and working, I think the suggestions can be moved into a followup PR.

Thank you for addressing those issues! I've added margin to the sides of the threads to prevent the button from sticking to the right side. Additionally, when the window is very small, the button now aligns under the channel name instead of disappearing.
<img width="202" alt="Screenshot 2024-06-03 at 14 36 47" src="https://github.com/ls1intum/Artemis/assets/69898283/c3374cd8-acb2-43ad-bea9-f7741bbabe1c">

edkaya reviewed (dismissed):
Reapprove

az108 reviewed (approved):
reapprove

Jan-Thurner reviewed (dismissed):
Thanks for implementing the feedback. Tested again on TS3. Works as expected.

sarpsahinalp reviewed (dismissed):
Reapprove

DominikRemo reviewed (changes requested):
Tested on ts4
I've noticed that opening a conversation always results in the tag 'new' being displayed for other users' messages, even if the message is not new and has already been viewed.

Please also have another look at the client tests.
`AnswerPostHeaderComponent › should set author information correctly` currently fails

beyzaaltuntas commented on src/main/webapp/app/shared/metis/post/post.component.ts:
```diff
@@ -107,6 +120,33 @@ export class PostComponent extends PostingDirective<Post> implements OnInit, OnC
+            this.sortedAnswerPosts = [];
+            return;
+        }
+        this.sortedAnswerPosts = this.posting.answers.sort(
```
Shouldn't we avoid in-place sorting?
 Maybe we can implement it like that:
` this.sortedAnswerPosts = this.posting.answers.slice().sort((a, b) => 
`

beyzaaltuntas commented on src/main/webapp/app/shared/metis/posting-footer/post-footer/post-footer.component.scss:
```diff
@@ -1,5 +1,10 @@
 @import 'src/main/webapp/app/shared/metis/metis.component';
 
+.reply-count {
+    font-size: 12px;
```
Don't we have shared font sizes somewhere?

beyzaaltuntas commented on src/main/webapp/app/shared/metis/post/post.component.ts:
```diff
@@ -107,6 +120,33 @@ export class PostComponent extends PostingDirective<Post> implements OnInit, OnC
+    /**
+     * sets the current post tags, empty error if none exit
+     */
+    private updateTags(): void {
+        if (this.posting.tags) {
+            this.tags = this.posting.tags;
+        } else {
+            this.tags = [];
+        }
+    }
```
You can shorten this to `this.tags = this.posting.tags ?? []`

egekurt123 commented:
> Tested on ts4 I've noticed that opening a conversation always results in the tag 'new' being displayed for other users' messages, even if the message is not new and has already been viewed.
> 
> Please also have another look at the client tests. `AnswerPostHeaderComponent › should set author information correctly` currently fails

Thanks for your comment and for pointing out the bug. Although this bug was not related to this PR, it has now been fixed. The test passes successfully as well.

DominikRemo reviewed (changes requested):
Thank you for fixing the client tests!
Tested on ts4.

The "New" badge does not appear at all on the communication page (intended?) and the problem persists in the messages page when switching between threads (the badges disappeared after reloading the page).

https://github.com/ls1intum/Artemis/assets/47261058/d9b0691b-bc74-4f32-b689-68874336466f

I have also noticed that the replies to a message are cut off in conversations. 
In the picture below, for example, there are only 3 replies, although there should be 9. 

![Bildschirmfoto 2024-06-16 um 21 53 07](https://github.com/ls1intum/Artemis/assets/47261058/fb44d819-58d1-4e2a-a171-d0bea0f8124b)

If there are a sufficient number of other messages, the responses cannot be expanded at all.

https://github.com/ls1intum/Artemis/assets/47261058/3c48357d-d381-4ff3-8871-5eebc6a60dc3

egekurt123 commented:
@DominikRemo Thanks for your reports.

- The "New" Badge only exists in the Messages Page
- I will take a look at the case without refreshing the page
- The UI Problem that causes messages to get cut off is a consequence of the deprecated paging in the communications page; my UI changes just apparently make it a lot worse. This is why I decided to merge this PR after #8801 which merges the Messages and Communication modules

az108 reviewed (dismissed):
reapprove after merge conflict

edkaya reviewed (approved):
reapprove after merge conflict

JohannesStoehr reviewed (changes requested):
The roles are assigned more or less randomly in the thread ui

SimonEntholzer reviewed (changes requested):
Tested during testing session, new changes look nice, only issue we found was that student/tutor tags over messages are not always shown correctly.

JohannesStoehr reviewed (approved):
Code with fix looks good to me now

Jan-Thurner reviewed (changes requested):
When writing messages as an intsructor, only the newest one is marked as an instructor. The other messages are marked as a student. When reloading the page, all messages are marked as instructor again.
![image](https://github.com/ls1intum/Artemis/assets/107639007/a8f768c7-79dd-44d5-8ee3-445eca1f1fbb)
After reload:
![image](https://github.com/ls1intum/Artemis/assets/107639007/78d302f9-2035-488f-bea3-fec15dd097be)

rabeatwork commented on src/main/webapp/app/shared/metis/posting-header/post-header/post-header.component.html:
```diff
@@ -1,25 +1,39 @@
-        <span class="posting-header header-author-date">
+<div class="d-flex justify-content-between my-1">
+    <div class="align-items-center">
+        <span class="fs-medium header-author-date">
```
See Feedback in src/main/webapp/app/shared/metis/posting-header/answer-post-header/answer-post-header.component.html

rabeatwork commented on src/main/webapp/app/shared/metis/posting-header/post-header/post-header.component.html:
```diff
@@ -1,25 +1,39 @@
-<div class="row justify-content-between my-1">
-    <div class="col-auto pe-0">
-        <span class="posting-header header-author-date">
+<div class="d-flex justify-content-between my-1">
```
Not part of this PR, but it would make sense to create a component for it. Since we use more or less the same code in several files.

rabeatwork commented on src/main/webapp/content/scss/global.scss:
```diff
@@ -1087,3 +1091,15 @@ body.transparent-background {
     display: none;
 }
+
+.fs-small {
```
Great idea to add global classes for these font sizes. However, please use either rem values or the corresponding keywords like `small` or `smaller` and so on. rem is relative to the font size of the root element. In our case 1 rem corresponds to 16px. The keywords correspond to the following rem values:

xx-small: 0.625rem
x-small: 0.75rem
small: 0.875rem
medium: 1rem
large: 1.125rem
x-large: 1.5rem
xx-large: 2rem
xxx-large: 3rem

rabeatwork commented on src/main/webapp/app/shared/metis/posting-header/post-header/post-header.component.html:
```diff
@@ -1,25 +1,39 @@
+            <span class="badge bg-secondary hideAfter5Seconds">{{ 'global.generic.new' | artemisTranslate }}</span>
+        }
+    </div>
+    <div class="mx-2 my-1">
```
Please make sure that these icons don't wrap 
<img width="345" alt="image" src="https://github.com/ls1intum/Artemis/assets/75392103/8b008906-e7b2-4f33-bca0-edf3b20fc65a">

egekurt123 commented on src/main/webapp/app/overview/discussion-section/discussion-section.component.scss:
```diff
@@ -2,7 +2,7 @@
 $draggable-width: 15px;
 $discussion-section-card-min-width-sm: 230px;
-$discussion-section-card-min-width: 360px;
+$discussion-section-card-min-width: 475px;
```
I changed it again, but the new font sizes and badges are causing an extra line break in the message header when users have longer names. We can look at it again and add a quick followup PR

egekurt123 commented on src/main/webapp/app/overview/course-conversations/layout/conversation-thread-sidebar/conversation-thread-sidebar.component.scss:
```diff
@@ -1,7 +1,7 @@
 
 $draggable-width: 15px;
-$thread-section-card-min-width: 420px;
+$thread-section-card-min-width: 500px;
```
I changed it again, but the new font sizes and badges are causing an extra line break in the message header when users have longer names. We can look at it again and add a quick followup PR

pzdr7 reviewed (commented):
Tested on TS5. 

I'm encountering an issue where the edit and delete icons don't match the messages in a thread.

To reproduce:

User 1 (instructor): Send two messages in a channel. On the first message, reply a few times.
User 2 (student): On the second message, reply a few times. Then, switch to the thread of the first message.

The delete/edit icons stay. In this example, I can see them as test user 3 even though the messages were written by test user 16.

![image](https://github.com/ls1intum/Artemis/assets/38403547/31e7c11b-6d75-4129-9d12-638aabf6e488)

(I assume it doesn't show up for the bottom one because the other thread, where I replied with test user 3, had just two replies instead of three).

https://artemis-test5.artemis.cit.tum.de/courses/20/communication?conversationId=226

egekurt123 commented:
> Tested on TS5.
> 
> I'm encountering an issue where the edit and delete icons don't match the messages in a thread.
> 
> To reproduce:
> 
> User 1 (instructor): Send two messages in a channel. On the first message, reply a few times. User 2 (student): On the second message, reply a few times. Then, switch to the thread of the first message.
> 
> The delete/edit icons stay. In this example, I can see them as test user 3 even though the messages were written by test user 16.
> 
> ![image](https://private-user-images.githubusercontent.com/38403547/346273213-31e7c11b-6d75-4129-9d12-638aabf6e488.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAyNzAyMjIsIm5iZiI6MTcyMDI2OTkyMiwicGF0aCI6Ii8zODQwMzU0Ny8zNDYyNzMyMTMtMzFlN2MxMWItNmQ3NS00MTI5LTlkMTItNjM4YWFiZjZlNDg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA2VDEyNDUyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczM2FhYTgyZmQyZTk1MjFmMzc2NDY4OTNhMTQ5NDEyNjNiYTUxNTVjZjhhZGM2MTVlODIyNDExZDA0ZGM5MzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YzDLCKOXHDFBetZ3eWDZDg3OuAKWn3KzomAvKRGOZ-8)
> 
> (I assume it doesn't show up for the bottom one because the other thread, where I replied with test user 3, had just two replies instead of three).
> 
> https://artemis-test5.artemis.cit.tum.de/courses/20/communication?conversationId=226

I have just took a deeper look into this and found that it can also be reproduced on other test servers, confirming that it is an existing bug.

JohannesStoehr reviewed (approved):
Code

Jan-Thurner reviewed (approved):
Thanks for looking into the issue and fixing it. LGTM now.

az108 reviewed (approved):
Changes LGTM

bassner reviewed (changes requested):
Please add the test coverage table to the PR

bassner commented:
@egekurt123 You did not remove the hardcoded colors 🫣 please read the docs i linked in my comment

bassner commented:
@egekurt123 You did not remove the hardcoded colors 🫣 please read the docs i linked in my comment
